File: webs/src/api/llm.js
import request from './request';

// LLM 整理节点
export function llmOrganizeNodes(data) {
  return request({
    url: '/v1/llm/organize-nodes',
    method: 'post',
    data,
    timeout: 120000
  });
}

// LLM 生成订阅规则
export function llmGenerateRules(data) {
  return request({
    url: '/v1/llm/generate-rules',
    method: 'post',
    data,
    timeout: 120000
  });
}

File: routers/llm.go
package routers

import (
	"sublink/api"
	"sublink/middlewares"

	"github.com/gin-gonic/gin"
)

func LLM(r *gin.Engine) {
	llmGroup := r.Group("/api/v1/llm")
	llmGroup.Use(middlewares.AuthToken)
	{
		llmGroup.POST("/organize-nodes", middlewares.DemoModeRestrict, api.LLMOrganizeNodes)
		llmGroup.POST("/generate-rules", middlewares.DemoModeRestrict, api.LLMGenerateRules)
	}
}
